docs/kyma/docs/037-gs-installation-overrides.md

```diff
@@ -0,0 +1,172 @@
+To customize Kyma installation, Installer exposes a generic mechanism for configuring Helm overrides called **user-defined** overrides.
+
+
+## User-defined overrides
+
+The Installer finds user-defined overrides by reading the ConfigMaps and Secrets deployed in the `kyma-installer` Namespace and marked with the `installer:overrides` Label.
+
+The Installer constructs a single override by inspecting the ConfigMap or Secret entry key name. The key name should be a dot-separated sequence of strings corresponding to the structure of keys in the chart's `values.yaml` file or the entry in chart's template. See the examples below.
+
+Installer merges all overrides recursively into a single YAML stream and passes it to Helm during the Kyma installation/upgrade operation.
+
+
+### Common vs component overides
+
+The Installer looks for available overrides each time a component installation or update operation is due.
+Overrides for the component are composed from two sets: **common** overrides and **component-specific** overrides.
+
+Kyma uses common overrides for the installation of all components. Only ConfigMaps or Secrets marked with the installer:overrides Label contain the definition of overrides (TODO: Fix it).
+
+Kyma uses component-specific overrides only for the installation of specific components. ConfigMaps and Secrets marked with both installer:overrides and component: <name> Labels, where <name> is the component name, contain the definition. Component-specific overrides have precedence over Common ones in case of conflicting entries.
+
+
+### Examples
+
+
+#### Top-level charts overrides
+
+Overrides for top-level charts are straightforward. Just use the template value from the chart (without leading ".Values." prefix) as the entry key in the ConfigMap or Secret.
+
+Example:
+
+The Installer uses a `core` top-level chart that contains a template with the following value reference:
+```
+memory: {{ .Values.test.acceptance.ui.requests.memory }}
+```
+The chart's default value `test.acceptance.ui.requests.memory` in the `values.yaml` file resolves the template.
+The following fragment of `values.yaml` shows this definition:
+```
+test:
+  acceptance:
+    ui:
+      requests:
+        memory: "1Gi"
+```
+
+To override this value, for example to "2Gi", proceed as follows:
+- Create a ConfigMap in the `kyma-installer` Namespace, labelled with: `installer:overrides` (or reuse an existing one).
+- Add an entry `test.acceptance.ui.requests.memory: 2Gi` to the map.
+
+Once the installation starts, the Installer generates overrides based on the map entries. The system uses the value of "2Gi" instead of the default "1Gi" from the chart `values.yaml` file.
+
+For overrides that the system should keep in Secrets, just define a Secret object instead of a ConfigMap with the same key and value. Don't forget to label the Secret with `installer:overrides`. Remember that values in Secrets must be base64-encoded. (TODO: same key and value, but base64...)
```

Comment:
There's a problem here. We should define a secret with the same key and value, but the value must be base64 so it's not the same, isn't it? Well, for the one who creates the object (Secret) it's different, because it's encoded base64. But for the _system_, for the software entities that will read this secret, this value _is_ the same, because Kubernetes automatically "unwraps" base64 encoding while reading Secrets.
We have to find a nice way to describe it. Perhaps this would be OK:
... just define a Secret object instead of a ConfigMap with the same key and a base64-encoded value.

Reply:
I agree.  

For overrides that the system should keep in Secrets, just define a Secret object instead of a ConfigMap with the same key and a base64-encoded value. Be sure to label the Secret with `installer:overrides`.